sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -50,10 +57,27 @@ export default function VisualSearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input) ?? '';
   const isLoading = useAnswersState(state => state.searchStatus.isLoading) ?? false;
-  const [executeQuery, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
+  const [executeQueryWithNearMeHandling, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
     return answersActions.executeUniversalAutocomplete();
   });
+  const recentSearches = useMemo(() => {
```

Comment:
is it possible to extract recentSearches to a hook?

Reply:
update: pr now use hook